test/direct-channel.test.js

```diff
@@ -22,255 +24,264 @@ const ipfsPaths = [
+      })
 
-      return new Promise(async (resolve, reject) => {
-        c1.on('error', reject)
-        c2.on('error', reject)
+      it('has an event emitter for \'message\' event', async () => {
+        let err
+        try {
+          c.on('message', () => {})
+        } catch (e) {
+          err = e
+        }
+        assert.equal(err, null)
+      })
+    })
 
-        c2.on('message', async (m) => {
-          assert.notEqual(m, null)
-          assert.equal(m.from, id1)
-          assert.equal(m.data.toString(), 'hello1')
-          assert.equal(m.topicIDs.length, 1)
-          assert.equal(m.topicIDs[0], c1.id)
-          assert.equal(m.topicIDs[0], c2.id)
-          await c2.send(Buffer.from('hello2'))
-        })
+    describe('messaging', function() {
+      it('sends and receives messages', async () => {
+        const c1 = await Channel.open(ipfs1, id2)
+        const c2 = await Channel.open(ipfs2, id1)
+
+        await c1.connect()
+        await c2.connect()
+
+        return new Promise(async (resolve, reject) => {
+          c1.on('error', reject)
+          c2.on('error', reject)
+
+          c2.on('message', async (m) => {
+            assert.notEqual(m, null)
+            assert.equal(m.from, id1)
+            assert.equal(Buffer.from(m.data).toString(), Buffer.from('hello1'))
+            assert.equal(m.topicIDs.length, 1)
+            assert.equal(m.topicIDs[0], c1.id)
+            assert.equal(m.topicIDs[0], c2.id)
+            await c2.send(Buffer.from('hello2'))
+          })
+
+          c1.on('message', (m) => {
+            assert.equal(m.from, id2)
+            assert.equal(Buffer.from(m.data).toString(), Buffer.from('hello2'))
+            assert.equal(m.topicIDs.length, 1)
+            assert.equal(m.topicIDs[0], c1.id)
+            assert.equal(m.topicIDs[0], c2.id)
+            c1.close()
+            c2.close()
+            setTimeout(() => resolve(), 500)
+          })
 
-        c1.on('message', (m) => {
-          assert.equal(m.from, id2)
-          assert.equal(m.data.toString(), Buffer.from('hello2'))
-          assert.equal(m.topicIDs.length, 1)
-          assert.equal(m.topicIDs[0], c1.id)
-          assert.equal(m.topicIDs[0], c2.id)
-          c1.close()
-          c2.close()
-          setTimeout(() => resolve(), 500)
+          await c1.send('hello1')
         })
-
-        await c1.send('hello1')
       })
     })
-  })
 
-  describe('connect', function() {
-    it('connects the peers', async () => {
-      let c1, c2
+    describe('connect', function() {
+      it('connects the peers', async () => {
+        let c1, c2
 
-      c1 = await Channel.open(ipfs1, id2)
-      c2 = await Channel.open(ipfs2, id1)
+        c1 = await Channel.open(ipfs1, id2)
+        c2 = await Channel.open(ipfs2, id1)
 
-      let peers = await ipfs1.pubsub.peers(c1.id)
-      assert.deepEqual(peers, [])
+        let peers = await ipfs1.pubsub.peers(c1.id)
+        // assert.deepEqual(peers, [])
```

Comment:
I don't want this line getting commented out to be missed. There seems to be a discrepancy between how `ipfs.pubsub.peers` behaves via the HTTP API and in-proc js-ipfs OR between go-ipfs and js-ipfs. Either way, when tested with go-ipfs it reports the id of ipfs2 in the `peers` array, not an empty array.

Reply:
Got it 👍  This may be that the peers connect (somehow) automatically at start already, which would explain this. Further, can't remember exactly why this assert was put here in the first place, but it does seem redundant for the purposes of the _actual_ test (make sure the peers are connected afte `connect()` was called).